dv/uvm/core_ibex/sim.py

```diff
@@ -190,6 +193,74 @@ def get_simulator_cmd(simulator, yaml_path, enables):
             subst_cmd(entry['sim']['cmd'], enables, entry['sim'], env_vars))
 
 
+def filter_tests_by_config(cfg, test_list):
+    '''Filter the provided testlist to ensure all required RTL parameters are
+       provided by the Ibex core configuration.
+
+    Arguments:
+
+        cfg: string name of the ibex config being tested, should match a
+             config name from ibex_configs.yaml.
+
+        test_list: list of test entry objects parsed from the YAML testlist
+
+    Returns:
+
+        filtered_test_list: a list of test entry objects, filtered such that
+                            all tests incompatible with the specified ibex
+                            config have been removed.
+
+                            e.g. if the "small" config has been specified, this
+                            function will filter out all tests that require
+                            B-extension and PMP parameters
+    '''
+    filtered_test_list = []
+    config = parse_config(cfg, os.path.join(_IBEX_ROOT, "ibex_configs.yaml"))
+
+    for test in test_list:
+        if "rtl_params" in test:
+            param_dict = test['rtl_params']
+            matching_params = False
+            for p in param_dict.keys():
+                # Throw an error if required RTL parameters in the testlist
+                # have been formatted incorrectly (typos, wrong parameters, etc)
+                if p not in config:
+                    logging.error('parameter % not found in config %' % (p, cfg))
+                    sys.exit(RET_FAIL)
+                else:
+                    config_val = config.get(p)
+                    p_val = param_dict.get(p)
```

Comment:
I'm not a fan of looking things up twice. There's a small performance cost (who cares!) but there's also a slight difficulty when reading: the reader has to look back to convince themselves the result exists. If you know that it does (like you do here), then `config[p]` is probably preferable to `config.get(p)` **but** you could do this more nicely still by iterating over `p, p_val in param_dict.items()` above and then moving the `config_val = config.get(p)` before the check on line 227, then checking whether `config_val is None` or not.

Reply:
sure, that sounds sensible to me, updated!